content/200-concepts/100-components/02-prisma-client/250-tracing.mdx

```diff
@@ -0,0 +1,213 @@
+---
+title: 'Tracing'
+metaTitle: 'Tracing (Preview)'
+metaDescription: 'Diagnose application performance with detailed traces of each query.'
+preview: true
+tocDepth: 3
+---
+
+<TopBlock>
+
+Prisma tracing logs
+Prisma tracing outputs
+
+Prisma tracing provides a detailed log of the activity that Prisma Client carries out, at a query level, including the time taken to execute each query. It helps you analyze your application's performance and identify bottlenecks. Tracing is fully compliant with [OpenTelemetry](https://opentelemetry.io/), so you can use it as part of your end-to-end application tracing path.
+
+<Admonition type="info">
+
+Tracing gives you a highly detailed, query-level insight into your Prisma project. If you want aggregated numerical reporting, such as query counts, connection counts, and total query execution times, see [Metrics](/concepts/components/prisma-client/metrics).
+
+</Admonition>
+
+</TopBlock>
+
+## About tracing
+
+When you enable tracing, Prisma Client writes out a trace for each request that it makes. **should we say operation here, rather than request**? Each trace includes one or more [spans](https://www.jaegertracing.io/docs/1.35/architecture/#span), each of which represents one stage in the request life-cycle (**check terminology here**), such as a database statement or command, or serialization and deserialization. Each span includes the round trip connection time from Prisma Client to the database and back.
+
+**Maybe a diagram here to show:**
+
+- Trace (Prisma operation)
+
+  - Span (such as a connection)
+  - Span (such as a database statement or command)
+  - Span (such as deserialization)
+
+  You can send tracing output to the console, or analyze it in any OpenTelemetry-compatible tracing system, such as [Jaeger](https://www.jaegertracing.io/).
+
+**TBA: maybe a screenshot in Jaeger with example**
+
+## Output
+
+Depending on the Prisma operation (**request?**), Prisma Client might output any of the following spans:
```

Comment:
One more to add here `'prisma:middleware'`. Represents time spent in the users defined [middleware](https://www.prisma.io/docs/concepts/components/prisma-client/middleware).

Reply:
Thanks - I'll add this.